services/geocoding.py

```diff
@@ -1,58 +1,69 @@
-import re
-import json
-import googlemaps
 from dataclasses import dataclass
-from typing import Callable, List, Mapping
-from returns.io import impure_safe
-from config.settings import Settings
-from models.point_of_interest import PointOfInterest
+from typing import Callable, List, Mapping, Set
 
+import googlemaps
 
+from config.settings import Settings
+from models.point_of_interest import PointOfInterest
 
 
 @dataclass
 class Point:
     lat: str
     lng: str
 
+
 def make_geocode_request(address: str, gmaps):
-   return gmaps.geocode(address)
+    return gmaps.geocode(address)
+
 
 def init_google_maps(key):
     return googlemaps.Client(key=key)
+
+
 @dataclass
 class GeoCodingProcessor:
 
     # Injected dependencies
     settings: Settings
+    log: Callable
     make_geocode_request: Callable = make_geocode_request
     init_google_maps: Callable = init_google_maps
-    
-    @impure_safe
+
     def enhance(self, entries: List[PointOfInterest]) -> List[PointOfInterest]:
         """
         For entries with missing lat/lng but present address
         we can find lat/lng using a geocoding API.
         """
 
-        #Create client instance of gmaps
+        # Create client instance of gmaps
         gmaps = self.init_google_maps(self.settings.developer_key)
 
-        addresses_to_geocode: List[str] = [
+        addresses_to_geocode: Set[str] = {
             poi.address
             for poi in entries
-            if not poi.lat or not poi.lng
-        ]
+            if (not poi.lat or not poi.lng)
+            and poi.address
```

Comment:
@wdunn001 I've switched to a list to set and check that address is not empty, otherwise, we've had  `addresses_to_geocode` like `['', '', '', '']`